src/server/lock_manager/waiter_manager.rs

```diff
@@ -253,10 +253,12 @@ impl WaiterManager {
             // because commit or rollback request will be sent to the new leader.
             //
             // `default_wait_for_lock_timeout` is the max timeout.
-            if timeout == 0 || timeout > self.default_wait_for_lock_timeout {
-                timeout = self.default_wait_for_lock_timeout;
+            if timeout == WaitTimeout::Default
```

Comment:
Could we define an indepenent function for `WaitTimeout` like `check_and_update(max_timeout)`？？

Reply:
Good idea!